src/blueapi/core/bluesky_types.py

```diff
@@ -76,7 +79,10 @@ def _follows_bluesky_protocols(obj: Any) -> bool:
 
 def is_bluesky_plan_generator(func: PlanGenerator) -> bool:
     try:
-        return get_type_hints(func).get("return") == MsgGenerator
+        return_type = get_type_hints(func).get("return")
+        return (get_origin(return_type) is ConcreteGenerator) and (
+            Msg in get_args(return_type)
```

Comment:
Should this be something like `get_args(return_type)[0] == Msg`? At the moment (I think) this will return true for something like `def foo() -> Generator[int, int, Msg]: ...` which I don't think counts as a plan?

Reply:
Thought of [some extra test cases](https://github.com/DiamondLightSource/blueapi/pull/798/commits/dd737df45869d8c764c403b020b5d402a6eb1754) that I was worried about in this case but they still pass, so changed